src/mmcache.hpp

```diff
@@ -241,7 +241,7 @@ if (cache[hidx].finger_print_cnt_sum <= 5)
       int shift = (int)minimizers[0].second >> 1;
       // Do not cache if it is too near the start.
       for (i = 0; i < size; ++i) 
-        if ((int)pos_candidates[i].position < 0) {
+        if ((int)pos_candidates[i].position < kmer_length + shift) {
```

Comment:
Why use kmer_length + shift? I thought it is just shift?

Reply:
Kmer_length+shift is safer. Just "shift" is not enough, it could be other read bearing the same minimizer vector has larger shift causing negative coordinate.

Reply:
I am worried about losing some candidates if this value is too large. Also can you explain if the minimizer vector shifts a lot in anther read while still being identical to a previous reads, we will still get negative candidates, which would cause problem?

Reply:
This will not lose candidates. It just means Chromap will not cache this minimizer vector at all. So for future queries, they always have cache miss and go through the index.